waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/CloseableThriftHiveMetastoreIfaceClientFactory.java

```diff
@@ -64,25 +65,31 @@ public CloseableThriftHiveMetastoreIfaceClientFactory(
 
   public CloseableThriftHiveMetastoreIface newInstance(AbstractMetaStore metaStore) {
     Map<String, String> properties = new HashMap<>();
+    String name = metaStore.getName().toLowerCase(Locale.ROOT);
     if (waggleDanceConfiguration.getConfigurationProperties() != null) {
       properties.putAll(waggleDanceConfiguration.getConfigurationProperties());
     }
     if (metaStore.getConfigurationProperties() != null) {
       properties.putAll(metaStore.getConfigurationProperties());
     }
+    CloseableThriftHiveMetastoreIface readWrite = null;
     if (metaStore.getGlueConfig() != null) {
-      return newGlueInstance(metaStore, properties);
+      readWrite = newGlueInstance(name, metaStore.getGlueConfig(), new HashMap<>(properties));
```

Comment:
Making copies of the properties map so when we reuse we don't accidentally add properties meant for readwrite